beep/tests/test_validate.py

```diff
@@ -210,13 +218,17 @@ def test_validation_maccor(self):
 class SimpleValidatorTest(unittest.TestCase):
     def setUp(self):
         # Setup events for testing
-        try:
-            kinesis = boto3.client('kinesis')
-            response = kinesis.list_streams()
-            self.events_mode = "test"
-        except Exception as e:
-            warnings.warn("Cloud resources not configured")
+        if not secret_accessible(ENVIRONMENT):
             self.events_mode = "events_off"
+        else:
+            try:
+                kinesis = boto3.client('kinesis')
+                response = kinesis.list_streams()
+                self.events_mode = "test"
+            except Exception as e:
+                warnings.warn("Cloud resources not configured")
+                self.events_mode = "events_off"
```

Comment:
Since we use this same code snippet multiple times, I recommend creating a method for it which can be called instead.

Reply:
Good thought -- pushing a new commit shortly.